include/rtc/rtp.hpp

```diff
@@ -0,0 +1,501 @@
+    inline uint32_t octetCount() const { return ntohl(_octetCount); }
+
+    inline void setNtpTimestamp(uint32_t ts) { _ntpTimestamp = htonll(ts); }
+    inline void setRtpTimestamp(uint32_t ts) { _rtpTimestamp = htonl(ts); }
+
+    inline void log() const {
+        header.log();
+        PLOG_VERBOSE << "RTCP SR: "
+                   << " SSRC=" << ntohl(senderSSRC) << ", NTP_TS=" << ntpTimestamp()
+                   << ", RTP_TS=" << rtpTimestamp() << ", packetCount=" << packetCount()
+                   << ", octetCount=" << octetCount();
+
+        for (unsigned i = 0; i < unsigned(header.reportCount()); i++) {
+            getReportBlock(i)->log();
+        }
+    }
+};
+
+struct RTCP_RR {
+    RTCP_HEADER header;
+    SSRC senderSSRC;
+
+private:
+    RTCP_ReportBlock _reportBlocks;
+
+public:
+    inline RTCP_ReportBlock *getReportBlock(int num) { return &_reportBlocks + num; }
+    inline const RTCP_ReportBlock *getReportBlock(int num) const { return &_reportBlocks + num; }
+
+    inline SSRC getSenderSSRC() const { return ntohl(senderSSRC); }
+    inline void setSenderSSRC(SSRC ssrc) { this->senderSSRC = htonl(ssrc); }
+
+    [[nodiscard]] inline size_t getSize() const {
+        // "length" in packet is one less than the number of 32 bit words in the packet.
+        return sizeof(uint32_t) * (1 + size_t(header.length()));
+    }
+
+    inline void preparePacket(SSRC senderSSRC, uint8_t reportCount) {
+        // "length" in packet is one less than the number of 32 bit words in the packet.
+        size_t length = (sizeWithReportBlocks(reportCount) / 4) - 1;
+        header.prepareHeader(201, reportCount, uint16_t(length));
+        this->senderSSRC = htonl(senderSSRC);
+    }
+
+    inline static size_t sizeWithReportBlocks(uint8_t reportCount) {
+        return sizeof(header) + 4 + size_t(reportCount) * sizeof(RTCP_ReportBlock);
+    }
+
+    inline void log() const {
+        header.log();
+        PLOG_VERBOSE << "RTCP RR: "
+                   << " SSRC=" << ntohl(senderSSRC);
+
+        for (unsigned i = 0; i < unsigned(header.reportCount()); i++) {
+            getReportBlock(i)->log();
+        }
+    }
+};
+
+
+struct RTCP_REMB {
+    RTCP_FB_HEADER header;
+
+    /*! \brief Unique identifier ('R' 'E' 'M' 'B') */
+    char id[4];
+
+    /*! \brief Num SSRC, Br Exp, Br Mantissa (bit mask) */
+    uint32_t bitrate;
+
+    SSRC ssrc[1];
+
+    [[nodiscard]] unsigned int getSize() const {
+        // "length" in packet is one less than the number of 32 bit words in the packet.
+        return sizeof(uint32_t) * (1 + header.header.length());
+    }
+
+    void preparePacket(SSRC senderSSRC, unsigned int numSSRC, unsigned int bitrate) {
+
+        // Report Count becomes the format here.
+        header.header.prepareHeader(206, 15, 0);
+
+        // Always zero.
+        header.setMediaSourceSSRC(0);
+
+        header.setPacketSenderSSRC(senderSSRC);
+
+        id[0] = 'R';
```

Comment:
In your example you moved this bit of the code to be part of line 314. I brought it back down here because I am unsure how I would trigger that. If you want to move this back up there could you also fix `RtcpReceivingSession::pushREMB`?

Reply:
Yes, good point, let's keep it here then :+1: